### DISCUSSION TITLE
comfortable mosaic view

### DISCUSSION THREAD

**Discussion by loulou91** [photoprism/photoprism#3582] [Q&A]
Is there a plan to implement a "comfortable" mosaic view as in other photos viewers as GooglePhotos or Librephotos?
I see at least 2 advantages, 1/ looks nice ;-) 2/ do not truncate photos!
Sorry il it already in the plan but I didn't found it!

**Comment by lastzero** [comment]
We've just improved it in our development preview. What else do you need? Of you want not tiles but the original aspect ratio, how would panoramas be displayed?

**Comment by loulou91** [comment]
As far as I see, the algorithm respects the ratio of photos but plays with zoom level of each individual photos to get an optimized visual result. Here is an example. It looks to cover panoramas case without any problem!
![screenshot](https://github.com/photoprism/photoprism/assets/51127123/00c01cd7-f5aa-4d9e-8f2f-54c559c1330a)
![image](https://github.com/photoprism/photoprism/assets/51127123/34f31ebf-bf1a-49d9-87ae-03f124ad160b)

**Reply by lastzero** [reply]
OK, so that's like https://github.com/schlosser/pig.js/ for which an issue might already exist. That's a different type of view than Mosaic which should consist of small and regular tiles based on its name.

**Reply by loulou91** [reply]
So it's looks not to be in your short term roadmap to implement this type of view (I don't know how to call it...)?
Should I raise an enhancement request?
https://feeding.schlosser.io/

**Reply by lastzero** [reply]
A related issue already exists. I was on my phone earlier and therefore couldn't search it for you:

- #85

As publicly announced, we have committed to first merging the existing pull requests (which includes making them releasable after merging) and then implementing OpenID Connect and 2FA. However, since this is an open source project, you or other contributors can help implement it in the short term! :)
